### PR TITLE
Added option to use (multiple) account(s) for athorization, while users

### PR DESCRIPTION
**Pull request by chris-andre**
A pull request regarding Issue #78 .

The plugin stays the same easy to use, but with "advanced" options to use accounts for authorization _if wanted_.

### CODE REVIEW

**Comment by dereuromark** [comment]
I am not sure if I like the added complexity of this edge case.
I am waiting for feedback from others here.

File: src/Auth/AclTrait.php [hunk 1]
```diff
@@ -345,20 +349,29 @@ protected function _getAvailableRoles() {
 	 * @throws \Cake\Core\Exception\Exception
 	 */
 	protected function _getUserRoles($user) {
+		$autorizeData = $user;
+
+		if ($this->getConfig('accountsTable') !== null) {
+			// If multipleAccounts === true, treat as indexed array and use index 0.
+			// Else, treat as associative and singularize the table name.
+			$autorizeData = $this->getConfig('multipleAccounts') === true ? $user[mb_strtolower($this->getConfig('accountsTable'))][0]
```
Comment (saeideng) [inline src/Auth/AclTrait.php]: why `$user[...][0]` ??

Comment (chris-andre) [reply src/Auth/AclTrait.php]: If multiple accounts is enabled, meaning that one user has (or may have) more than one account associated (which is kind of the meaning with a user-authenticate/account-authorize), the user should be allowed to use _one_ account at time.

Developer should make sure $user[...][0] is the active account when containing account in the user session. 

In my test app I use:
```
public function findAuth(Query $query, array $options)
    {
        $query
            ->contain(['Accounts' => function ($q) {
                return $q
                ->contain(['Roles', 'Tenants'])
                ->where(['Accounts.active_account' => true]);
            }])
            ->where(['Users.active' => 1]);

        return $query;
    }
```
With this finder only `Accounts.active_account => true` is selected, where I make sure each user only have **one** active account at the time. If no account is active, user is forced to select an account.

Now, lets say this is a tenant system. joe@doe.com is a user who has two accounts. One account is associated to tentant CakePhp and one account is associated to tenant GitHub. CakePhp account has Admin role, while GitHub account has User role. You will make sure the user/account has the role(s) that belongs to the tenant-scope you are in at the moment. You won't let the user have Admin role while navigating in the GitHub account/scope.

Comment (chris-andre) [reply src/Auth/AclTrait.php]: To add _the simple explaination_: `multipleAccounts` should be true when users hasMany accounts, and false when users hasOne accounts. Still, if users hasMany accounts, only one account's role(s) should be fetched at the time of running.

**Comment by dereuromark** [comment]
@ everyone
Is there any way we could enable this without having to hardcode it? A SOLID (mainly "closed for modification, open for extension" approach that allows hooking this in without the need to merge this very specific multi tenant kind of opt-in?

**Comment by saeideng** [comment]
>Developer should make sure $user[...][0] is the active account when containing account in the user session.

I say `$user` contains just one user data , not more

**Comment by chris-andre** [comment]
@saeideng It's a valid point. A more predictable approach.

**Pull request closed without merging**